src/services/gcs/backend.rs

```diff
@@ -0,0 +1,718 @@
+                debug!("object {} create finished", args.path());
+                Ok(())
+            }
+            _ => {
+                let err =
+                    parse_error_response("create", args.path(), parse_error_status_code, resp)
+                        .await;
+                warn!("object {} create: {:?}", args.path(), err);
+                Err(err)
+            }
+        }
+    }
+    #[trace("read")]
+    async fn read(&self, args: &OpRead) -> Result<BytesReader> {
+        increment_counter!("opendal_gcs_read_requests");
+
+        let p = self.get_abs_path(args.path());
+        debug!(
+            "object {} read start: offset {:?}, size {:?}",
+            &p,
+            args.offset(),
+            args.size()
+        );
+
+        let resp = self
+            .get_object(&p, args.offset(), args.size())
+            .await
+            .map_err(|e| {
+                error!("object {} get_object: {:?}", p, e);
+                e
+            })?;
+
+        match resp.status() {
+            StatusCode(x) if 200 <= x && x < 300 => {
+                debug!(
+                    "object {} reader created: offset {:?}, size {:?}",
+                    &p,
+                    args.offset(),
+                    args.size()
+                );
+
+                Ok(Box::new(resp.into_body()))
+            }
+            _ => {
+                let err =
+                    parse_error_response("read", args.path(), parse_error_status_code, resp).await;
+                warn!("object {} read: {:?}", args.path(), err);
+                Err(err)
+            }
+        }
+    }
+
+    #[trace("write")]
+    async fn write(&self, args: &OpWrite) -> Result<BytesWriter> {
+        let p = self.get_abs_path(args.path());
+        debug!("object {} write start: size {}", &p, args.size());
+
+        let (tx, body) = new_http_channel(args.size());
+
+        let req = self.insert_object(&p, args.size(), body).await?;
+
+        let bs = HttpBodyWriter::new(
+            args,
+            tx,
+            self.client.send_async(req),
+            HashSet::from([StatusCode::CREATED, StatusCode::OK]),
+            parse_error_status_code,
+        );
+
+        Ok(Box::new(bs))
+    }
+
+    #[trace("stat")]
+    async fn stat(&self, args: &OpStat) -> Result<ObjectMetadata> {
+        increment_counter!("opendal_gcs_stat_requests");
+
+        let p = self.get_abs_path(args.path());
+        debug!("object {} stat start", &p);
+
+        // Stat root always returns a DIR.
+        if self.get_rel_path(&p).is_empty() {
+            let mut m = ObjectMetadata::default();
+            m.set_mode(ObjectMode::DIR);
+
+            debug!("backed root object stat finished");
+            return Ok(m);
+        }
+
+        let resp = self.get_object_metadata(&p).await?;
+
+        match resp.status() {
+            StatusCode(code) if 200 <= code && code < 300 => {
+                let mut m = ObjectMetadata::default();
```

Comment:
I believe all code below is wrong.

Please parse the json returned by gcs: https://cloud.google.com/storage/docs/json_api/v1/objects#resource